### ISSUE TITLE
x/pkgsite: support viewing symbols from newer Go versions than declared in the module

### ISSUE THREAD

**Issue by codyoss** [golang/go#68979]
### What is the URL of the page with the issue?

https://pkg.go.dev/cloud.google.com/go/secretmanager@v1.14.0/apiv1#SecretIterator

### What is your user agent?

Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36

### What did you do?

I tried to view the docs of my newly released package for code gated by a buildtag: https://github.com/googleapis/google-cloud-go/blob/f5833e6cd8d2fc39e4decb9962062904601fc9c9/secretmanager/apiv1/auxiliary_go123.go#L17-L39

### What did you see happen?

It appears that only docs are shown for the version declared in the go.mod: go 1.21: https://github.com/googleapis/google-cloud-go/blob/f5833e6cd8d2fc39e4decb9962062904601fc9c9/secretmanager/go.mod#L3

### What did you expect to see?

I expect users of the library to be able to see docs for all Go versions or have the ability to select a Go version to see the docs of.

As a library maintainer it is important to me to maintain broad support for versions of Go. But it is also important for me to be able to implement and share new language features like iterator support gated by build tags. 

For standard library packages it seems you can see all supported symbols, and it even lets you know when things were added. Example: https://pkg.go.dev/net/http#Request.PathValue

**Comment by codyoss** [comment]
Closing in favor of #68932 as it is an identical issue.
